workshops/models.py

```diff
@@ -227,9 +228,14 @@ def ongoing_events(self):
         return queryset
 
     def unpublished_events(self):
-        '''Return a queryset for events that are not yet published.'''
+        '''Return events without URLs that are upcoming or have unknown starts.
 
-        return self.filter(published=False)
+        Events are ordered by slug and then by serial number.'''
+
+        future_without_url = (Q(start__gte=datetime.date.today()) & Q(url__isnull=True))
+        unknown_start = Q(start__isnull=True)
+        return self.filter(future_without_url | unknown_start)\
+                   .order_by('slug', 'id')
```

Comment:
From [PEP 8][1], recommended wrapping is by parentheses, brackets and braces, so maybe:

    return (self.filter(future_without_url | unknown_start)
                .order_by('slug', 'id'))

[1]: https://www.python.org/dev/peps/pep-0008/#maximum-line-length